internal/signal/signal.go

```diff
@@ -21,30 +18,56 @@ var exitSignals = []os.Signal{
 	syscall.SIGTERM,
 }
 
-// Stopper is an interface for anything with a Stop method.
-type Stopper interface {
-	Stop()
+// Handler is a simply a function intended to be called in response to a signal.
+type Handler func()
+
+// Exit reifies the idea of "an exit signal" so that such an exit is smart: It
+// allows you to add handlers that will be called when it occurs (AddHandler),
+// and to block until it does occur (Await).
+type Exit interface {
+	AddHandler(Handler)
```

Comment:
This name isn't descriptive enough - it will be invoked as `signal.AddHandler()` and doesn't describe well enough what signal and on what event it will be triggered.

Reply:
The current code looks like this:

```go
exitSignal := signal.NewExit()
exitSignal.AddHandler(...)
```
I think this looks good and is in line with Go conventions.  Ie, use the package name as a descriptive modifier so your package types and funcs don't stutter, and then correctly name your local variables to capture the context.